packages/block-editor/src/components/iframe/index.js

```diff
@@ -174,9 +174,16 @@ function Iframe(
 	const setRef = useRefEffect( ( node ) => {
 		function setDocumentIfReady() {
 			const { contentDocument, ownerDocument } = node;
-			const { readyState, documentElement } = contentDocument;
-
-			if ( readyState !== 'interactive' && readyState !== 'complete' ) {
+			const { readyState, documentElement, compatMode } = contentDocument;
+
+			// As srcDoc loads contents asynchronously this will cause the iframe to
+			// load documents twice. We need to hook react to the correct contentDocument
```

Comment:
When you're testing, does the document really load twice for you? With `srcDoc`, it should only load once.

When I run this little test app:
```js
function App() {
  const elRef = React.useRef();
  React.useEffect(() => {
    function onLoad() {
      const doc = elRef.current.contentDocument;
      console.log("load", doc.readyState, doc.compatMode);
    }
    elRef.current.addEventListener("load", onLoad);
    return () => elRef.current.removeEventListener("load", onLoad);
  }, []);

  return <iframe ref={elRef} srcDoc="<!doctype html>" />;
}
ReactDOM.render(<App />, document.getElementById("root"));
```
I get only one `load` event logged:
```
load complete CSS1Compat
```

Reply:
Double checked by logging and breaking to inspect the iframe node and this is what we get:
![Screenshot 2022-03-08 at 10 51 42](https://user-images.githubusercontent.com/112270/157212319-8f297f2b-1265-434c-9c9c-8129476de131.jpg)

Reply:
I'm thinking if it wasn't because first load fires before ref is set and useEffect fires?

Reply:
Or more likely it's a browser difference - Found this exploration https://stackoverflow.com/questions/10781880/dynamically-created-iframe-triggers-onload-event-twice/38459639#38459639 it seems it might be Chrome/webkit only?

Reply:
On Firefox I get:
![Screenshot 2022-03-08 at 11 15 01](https://user-images.githubusercontent.com/112270/157216135-9107c4bf-3290-4900-a52d-11f9035d93d6.jpg)

Reply:
On my minimal example, I get only one load event even for Firefox: `load complete CSS1Compat`. I'll try it with the full Gutenberg editor.

> I'm thinking if it wasn't because first load fires before ref is set and useEffect executes?

Yes, the load certainly fires right after the element is added to DOM, which is before refs are set and effects executed. But the `iframe` element has the `srcDoc` attribute from the very beginning, and therefore should load only once.

Reply:
Ok got it. First one is initial state not `onload` event. It is not triggered by the listener but by the manual call to the setDocumentIfReady handler. 

https://github.com/WordPress/gutenberg/blob/21e2d2249e8a478fd4c6b49e44c1c170e5c44b2d/packages/block-editor/src/components/iframe/index.js#L213-L220

We have the comment here that load happens only in `firefox` but now with srcDoc in place it will fire everywhere it seems.

Should we remove L213-215 and leave only the onload event? This way we could remove or simplify the comment and ready check before we initialise the iframe content.

Reply:
> Should we remove L213-215 and leave only the onload event?

Yes, it's most likely guaranteed that the load is async, and the `useRefEffect` callback is always called before the load finishes. We can rely on the `load` event to fire and avoid checking synchronously.

Reply:
Yes, probably good to only use the load event now. The reason we didn't use the load event for all browsers was because we could start loading the editor much quicker.

Reply:
I've removed the comment and direct call and removed the `compatMode` check from ready check. I believe with how much we've simplified it the simple comment around iframe event binding and reason behind srcDoc with E2E tests explaining it a bit more and guarding for ensuring we have the right `compatMode` should be enough.